assets/js/application.js

```diff
@@ -2,22 +2,30 @@ require("expose-loader?$!expose-loader?jQuery!jquery");
 require("bootstrap-sass/assets/javascripts/bootstrap.js");
 
 import Vue from "vue";
-import VueRouter from "router";
-Vue.use(VueRouter);
 
-import AppComponent from "./components/app.vue";
-import BandComponent from "./components/band.vue";
+/**
+ *  Since we're using the runtime build of Vue, we're going to have to
+ *  pass it to the window for the router to see it. Nevertheless, it 
+ *  cleans up this file nicely.
+ */
 
-const routes = [
-  {path: "/band/:id", component: BandComponent, name: "showBand"},
-  {path: "/", component: AppComponent}
-];
+window.Vue = Vue;
 
-const router = new VueRouter({
-  mode: "history",
-  routes
-});
+/**
+ * Pass axios to the window, that way all our .vue files can
+ * see it and use it freely without having to import to each file.
+ * 
+ * Axios also has really really nice documentation to help you out!
+ * It may look slightly difference to what I've written in the vue files,
+ * because I've used ES6 syntax.
+ * 
+ * https://github.com/axios/axios
+ */
 
-const app = new Vue({
-  router
-}).$mount("#app");
+window.axios = require('axios');
```

Comment:
Since this is suppose to be just an example, I'm sure how 'proper' it's suppose to be...but aren't global window variables discouraged (https://vuejsdevelopers.com/2017/04/22/vue-js-libraries-plugins/)? I'm not sure a plugin is the right answer for something like this either. Maybe something in between, like:
```import axios from 'axios';
Object.definePrototype(Vue.prototype, '$http', { value: axios });```

Reply:
This is purely an example, I can see why window variables are discouraged but with the issues explained in that article i.e putting the app to a server will return an error (I've never experienced this personally.)

This is just my take on the vue part of buffalo and my personal experiences I've always, always, always separated the router from the app.js to clean things up. 

As I stated to Mark before making this PR, this is MY take on the vue side of things and it's completely open for discussion and this is how i'd work on projects using the VueRouter as in this example it is in fact an extra and not required with a Vue project.

But again, everything is open for debate, that's how things get better!

Reply:
Another reason I forgot to mention, if Golang developers are new to vue, perhaps they've come from a different framework I feel what I've fleshed together should be very nice building blocks to get them started in a comfortable vue environment.